openml/datasets/dataset.py

```diff
@@ -169,13 +169,13 @@ def __init__(self, name, description, format, dataset_id=None,
                                    for name, type_ in data['attributes']]
                     attribute_names = [name for name, type_ in data['attributes']]
 
-                    if isinstance(data['data'], tuple):
+                    if format.lower() == 'sparse_arff':
```

Comment:
Is it formally defined that in case of 'dense' the format is stored as 'arff' and in case of spare the format is stored as 'sparse_arff'?

At least, I am not aware of this being tested. 

@joaquinvanschoren what do you think?

Reply:
I thought that's what the field from OpenML is good for. Actually, the code is already depending on this (https://github.com/openml/openml-python/blob/781815ba036905e74c43f114f5d0e193435260d9/openml/datasets/dataset.py#L172).